Open resource card

cryogenian opened the pull request:
![open_resource_card](https://cloud.githubusercontent.com/assets/10245930/14643542/70ae7f12-0657-11e6-8ca6-1e8acc8d8f53.gif)

Open resource card
+ Removed `Explore` card 
+ Unified `initialStates` of cards 
+ `OpenResource` card
+ Updated tests 
+ A bit more unicode 

`FileSystem.Listing` has much more complicated interface and I couldn't find what can be extracted. 

@jonsterling please, review

jonsterling commented:
Thanks @cryogenian! I will review this this morning.

jonsterling commented on test/src/Test/SlamData/Feature/Interactions.purs:
```diff
@@ -189,9 +192,33 @@ playLastCard ∷ SlamFeature Unit
-provideFileInLastExploreCard =
-  provideFieldValue $ XPath.last $ XPath.anywhere XPaths.exploreInput
+selectResourceByPath ∷ String → SlamFeature Unit
+selectResourceByPath p = do
```
@beckyconning Can you comment on the naming of this function? It looked fine to me at first glance, but I know that you have designed a scheme for such functions.

beckyconning replied:
thanks for pinging me. 

the only information presented is the name of the card "Explore" and the tooltip on the dropdown button which is "Show file list". as such i think the interaction as it stands is "selecting a file to explore" and it refers to the last card.

so maybe something like `selectFileToExploreInLastCard`?

beckyconning replied:
Actually the function refers to the last explore card rather than the last card so

`selectFileToExploreInLastExploreCard`

or

`selectFileForLastExploreCard`

beckyconning replied:
If resource is a more accurate word and we add a label for the input that says something like "Select resource" and change the tooltip to "Open resource list" then `selectResourceByPathForLastExploreCell` or `selectLastExploreResourceByPath` would be fine by me : ).

cryogenian replied:
I'll use 'selectFileForLastExploreCard', it's the shortest and describes action precisely enough :)

jonsterling commented on src/SlamData/Notebook/Card/OpenResource/Component.purs:
```diff
@@ -0,0 +1,214 @@
+      H.modify (_selected ?~ r)
+    Right dp → do
+      H.modify (_browsing .~ dp)
+      H.modify (_selected .~ Nothing)
```
It probably doesn't matter too much, but what about collapsing these two calls into a single transaction, by composing the two lens updates?

cryogenian replied:
I'm fine with both approaches, maybe it's ok to remove lenses and use record update, what do you think?

jonsterling replied:
I have no problem with removing the use of lenses, though I would just say that the benefit is that they are a bit more modular than record update—for instance, if we wanted to, we could make the `State` type abstract, or implement it in a different way, without disturbing code that uses it.

As for whether to collapse into a single transaction, I guess there are two questions that should be considered:

1. does each call to `modify` result in a re-render? 
2. if so, does either of these updates make sense without the other?

jonsterling commented:
@cryogenian Modulo the things that we discussed in the thread above, this PR looks *awesome*! Really nice work as always.

cryogenian commented:
Thank you!

+ composed `H.modify` functions
+ renamed `selectResourceByPath` -> `selectFileForLastExploreCard`
+ renamed `resourceOpened` -> `resourceOpenedInLastExploreCard`

jonsterling commented:
great, thanks! Looks ready to go 😄 

👍
